src/lightning/fabric/utilities/load.py

```diff
@@ -0,0 +1,203 @@
+            old_lt = data._load_tensor
+
+            def _load_tensor() -> Parameter:
+                t = old_lt()
+                return torch._utils._rebuild_parameter(t, requires_grad, backward_hooks)
+
+            data._load_tensor = _load_tensor  # type: ignore[assignment]
+            return data
+        return torch._utils._rebuild_parameter(data, requires_grad, backward_hooks)
+
+    @classmethod
+    def rebuild_tensor_v2(
+        cls,
+        storage: "TypedStorage",
+        storage_offset: int,
+        size: tuple,
+        stride: tuple,
+        requires_grad: bool,
+        backward_hooks: OrderedDict,
+        metadata: Optional[Any] = None,
+        *,
+        archiveinfo: "_LazyLoadingUnpickler",
+    ) -> "_NotYetLoadedTensor":
+        rebuild_args = (storage_offset, size, stride, requires_grad, backward_hooks, metadata)
+        metatensor = torch._utils._rebuild_tensor_v2(
+            storage, storage_offset, size, stride, requires_grad, backward_hooks, metadata
+        )
+        storageinfo = storage.archiveinfo
+        return _NotYetLoadedTensor(metatensor, archiveinfo, storageinfo, rebuild_args)
+
+    def _load_tensor(self) -> Tensor:
+        from torch.storage import TypedStorage, UntypedStorage
+
+        name, storage_cls, fn, device, size = self.storageinfo
+        dtype = self.metatensor.dtype
+
+        storage = self.archiveinfo.file_reader.get_storage_from_record(
+            f"data/{fn}", size * torch._utils._element_size(dtype), UntypedStorage
+        )
+        uts = storage._typed_storage()._untyped_storage
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            storage = TypedStorage(wrap_storage=uts, dtype=dtype, _internal=True)
+        return torch._utils._rebuild_tensor_v2(storage, *self.rebuild_args)
+
+    @classmethod
+    def __torch_function__(
+        cls,
+        func: Callable,
+        types: Sequence,
+        args: Sequence[Any] = (),
+        kwargs: Optional[Dict] = None,
+    ) -> Any:
+        kwargs = kwargs or {}
+        loaded_args = [(arg._load_tensor() if isinstance(arg, _NotYetLoadedTensor) else arg) for arg in args]
+        return func(*loaded_args, **kwargs)
+
+    def __getattr__(self, name: str) -> Any:
+        # These properties don't require materialization and can be accessed through the meta tensor directly
+        if name in {
+            "dtype",
+            "grad",
+            "grad_fn",
+            "layout",
+            "names",
+            "ndim",
+            "output_nr",
+            "requires_grad",
+            "retains_grad",
+            "size",
+            "shape",
+            "volatile",
+        }:
+            return getattr(self.metatensor, name)
+
+        # TODO: needed for us?
+        # materializing with contiguous is needed for quantization
+        if name in {"contiguous"}:
+            return getattr(self._load_tensor(), name)
```

Comment:
I should probably keep this here if we want to import and use the util in lit-gpt?

Reply:
It might also be a generally applicable limitation. On the other hand, we could do this manually in the quantization code directly.

What do you suggest @t-vi?